### PR TITLE
treeview: recursive folding/unfolding with shift key held down

### PR DESCRIPTION
**Pull request by manavortex**
in tree view, opening/closing a node with shift key will also affect its children. 

Implemented extra checks for some objects in mesh/app, as these are the ones where not having it annoyed me the most. 
Feedback welcome, I have actually no idea what the correct way is of getting this helper class into the model.

### CODE REVIEW

File: WolvenKit.App/ViewModels/Tools/ChunkViewModelTools.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using DynamicData.Binding;
+using WolvenKit.App.ViewModels.Shell;
+using WolvenKit.RED4.Types;
+
+namespace WolvenKit.App.ViewModels.Tools;
+
+public class ChunkViewModelTools
```
Comment (rfuzzo) [inline WolvenKit.App/ViewModels/Tools/ChunkViewModelTools.cs:10]: public static class ChunkViewModelExtensions

File: WolvenKit.App/ViewModels/Tools/ChunkViewModelTools.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+{
+    public void SetChildExpansionStates(ChunkViewModel? node, bool isExpanded)
```
Comment (rfuzzo) [inline WolvenKit.App/ViewModels/Tools/ChunkViewModelTools.cs:12]: public static void SetChildExpansionStates(this ChunkViewModel? node, bool isExpanded)

File: WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs [hunk 3]
```diff
@@ -49,6 +50,7 @@ namespace WolvenKit.App.ViewModels.Shell;
 public partial class ChunkViewModel : ObservableObject, ISelectableTreeViewItemModel, INode<ReferenceSocket>
 {
     private readonly IChunkViewmodelFactory _chunkViewmodelFactory;
+    private readonly ChunkViewModelTools _chunkViewModelTools = new ChunkViewModelTools();
```
Comment (rfuzzo) [inline WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs:53]: delet

File: WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs [hunk 4]
```diff
@@ -196,6 +201,11 @@ public partial class ChunkViewModel : ObservableObject, ISelectableTreeViewItemM
     {
         if (IsExpanded && !_propertiesLoaded)
             CalculateProperties();
+
+        if (IsShiftBeingHeld)
+        {
+            _chunkViewModelTools.SetChildExpansionStates(this, IsExpanded);
```
Comment (rfuzzo) [inline WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs:207]: this.SetChildExpansionStates(IsExpanded);

**Review by rfuzzo — CHANGES_REQUESTED** [review]
could do it like this I guess, with extension methods.

File: WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs (hunk 4 above)
Comment (manavortex) [reply WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs:207]: Is it enough to have <ClassName>Extension.cs to have it immediately available?

Comment (rfuzzo) [reply WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs:207]: doesn't depend on the name, it is available when you have a static method with this as first argument for that class

Comment (manavortex) [reply WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs:207]: Ahh, yeah - I wanted to make it a static class, but that would make HotReload not work for it, so I instantiated it. Any advice about that?

Comment (rfuzzo) [reply WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs:207]: huh, I didn't know that... no, no idea. that's weird but okay we can leave it nonstatic then I guess

Comment (manavortex) [reply WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs:207]: I don't want to instantiate this for every single tree view node that I'm touching >.< I'll check later if I can make a factory class or something

**Review by rfuzzo — APPROVED** [review]

**Pull request merged**